bundler/lib/bundler/fetcher.rb

```diff
@@ -148,14 +148,11 @@ def specs(gem_names, source)
       specs.each do |name, version, platform, dependencies, metadata|
         spec = if dependencies
           EndpointSpecification.new(name, version, platform, self, dependencies, metadata).tap do |es|
-            unless index.local_search(es).empty?
```

Comment:
I'm not sure about removing this, but it seems to me that << just overwrites so we can too. The case not covered here is replacing one that has a checksum with one that doesn't.